phaser/phase.py

```diff
@@ -1,13 +1,153 @@
+from abc import ABC, abstractmethod
+
 import pandas as pd
 import logging
 from .column import make_strict_name, Column
-from .pipeline import Pipeline, Context, DropRowException, WarningException, PipelineErrorException
-from .steps import ROW_STEP, BATCH_STEP, DATAFRAME_STEP, PROBE_VALUE, row_step
+from .pipeline import Pipeline, Context, DropRowException, WarningException, PipelineErrorException, PhaserException
+from .steps import ROW_STEP, BATCH_STEP, PROBE_VALUE, row_step
 
 logger = logging.getLogger('phaser')
 logger.addHandler(logging.NullHandler())
 
-class ReshapePhase:
+
+class PhaseBase(ABC):
+
+    def __init__(self, name, context=None, error_policy=None):
+        self.name = name or self.__class__.__name__
+        self.context = context or Context()
+        self.error_policy = error_policy or Pipeline.ON_ERROR_COLLECT
+        self.headers = None
+        self.row_data = None
+
+    @abstractmethod
+    def run(self):
+        pass
+
+    def read_csv(self, source):
+        """ Includes the settings phaser normally uses with pandas read_csv """
+        return pd.read_csv(source,
+                         dtype='str',
+                         sep=',',
+                         skip_blank_lines=True,
+                         index_col=False,
+                         comment='#')
+
+    def load(self, source):
+        """ When creating a Phase, it may be desirable to subclass it and override the load()
+        function to do a different kind of loading.  Be sure to load the row_data into the
+        instance's row_data attribute as an iterable (list) containing dicts.
+        Defaults:
```

Comment:
Should this explanation of the default settings for reading csv be on the `read_csv` method? Then if someone wants to change only how csv is read in, that one method would be documented correctly.